org.mixxx.Mixxx.yaml

```diff
@@ -219,76 +132,79 @@ modules:
-        commands:
-          - cp -p /usr/share/automake-*/config.{sub,guess} autoaux
-          - autoreconf -si
-  
-  - name: libmodplug
+      - type: archive
+        url: https://github.com/jiixyj/libebur128/archive/refs/tags/v1.2.6.tar.gz
+        sha256: baa7fc293a3d4651e244d8022ad03ab797ca3c2ad8442c43199afe8059faa613
+
+  - name: libkeyfinder
+    buildsystem: cmake-ninja
+    # BUILD_TESTING=ON tries to download catch2 during the build step
     config-opts:
-      - --with-pic
+      - -DBUILD_TESTING=OFF
     sources:
       - type: archive
-        url: http://downloads.sourceforge.net/project/modplug-xmms/libmodplug/0.8.8.5/libmodplug-0.8.8.5.tar.gz
-        sha256: 77462d12ee99476c8645cb5511363e3906b88b33a6b54362b4dbc0f39aa2daad
+        url: https://github.com/mixxxdj/libkeyfinder/archive/refs/tags/v2.2.5.tar.gz
+        sha256: 516570e310f5bb5d87146fbefb129eb972dab3347987783554001e2cac26d9d6
 
   - name: mixxx
-    buildsystem: simple
-    build-commands:
-      - scons prefix=${FLATPAK_DEST} -j ${FLATPAK_BUILDER_N_JOBS} build=release ffmpeg=1 faad=1
-      - scons prefix=${FLATPAK_DEST} install
-      - install -Dm644 ${FLATPAK_DEST}/share/pixmaps/mixxx_icon.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+    buildsystem: cmake-ninja
+    # USE_SYMLINKS is only required for running the tests and fails with:
+    #
+    # [70/835] Symlinking test data to build directory...
+    # FAILED: CMakeFiles/mixxx-testdata
+    # cd /run/build/mixxx && /usr/bin/cmake -E create_symlink /run/build/mixxx/src/test /run/build/mixxx/src/test
+    # failed to create symbolic link '/run/build/mixxx/src/test' because existing path cannot be removed: Is a directory
+    #
+    # Unfortunately there is no way for a Flatpak to install udev rule files, so that needs to be installed manually
+    # by users who have HID or USB Bulk controllers.
+    config-opts:
+      - -DUSE_SYMLINKS=OFF
+      - -DINSTALL_USER_UDEV_RULES=OFF
+    post-install:
+      - mv ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/mixxx.svg
+        ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
       - icon_in="${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg";
         icon_out="${FLATPAK_ID}.png";
         for s in {32,64,128,256,512}; do
-        rsvg-convert "${icon_in}" -w "${s}" -h "${s}" -a -f png -o "${icon_out}";
-        install -p -Dm644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/";
+          rsvg-convert "${icon_in}" -w "${s}" -h "${s}" -a -f png -o "${icon_out}";
+          install -p -Dm644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/";
         done;
-    post-install:
-      - desktop-file-edit --set-key="Exec" --set-value="mixxx %U" ${FLATPAK_DEST}/share/applications/mixxx.desktop
       - desktop-file-edit --set-key="Icon" --set-value=${FLATPAK_ID} ${FLATPAK_DEST}/share/applications/mixxx.desktop
-      - install -d /app/extensions/Plugins
```

Comment:
This will cause the flatpak to not launch.

Reply:
Case in point, with the debug build:
```
bwrap: Can't mkdir parents for /app/extensions/Plugins: Read-only file system
```

Reply:
Huh? Mixxx has no plugins.

Reply:
Oh is this for the org.freedesktop.LinuxAudio.Plugins LV2 plugins?

Reply:
![works-on-my-machine](https://user-images.githubusercontent.com/9455094/129484080-53cb9c9f-3ead-4bcc-9151-e7783d63311e.png)

Reply:
https://dl.flathub.org/build-repo/54197/org.mixxx.Mixxx.flatpakref runs on my machine but no LV2 plugins are detected.

Reply:
so basically your are telling the person that actually has the expertise on the topic that they are wrong?

I actually tested https://github.com/flathub/org.mixxx.Mixxx/pull/33#discussion_r689103753

Reply:
I'm not telling you that you're wrong. I'm saying I tested it on my computer and didn't reproduce the error you encountered. I don't know why we're having different results.

Reply:
I also did not encounter that error with my local builds.

Reply:
Can you explain what the purpose of that removed line was? Was it for LV2 plugins, or the VAMP and SoundSource plugins that Mixxx used to have before 2.3, or both?

Reply:
Wait, is that error you pasted above from a build log or something you see running the Flatpak?

Reply:
The LV2 path is set at https://github.com/flathub/org.mixxx.Mixxx/blob/b4235f1c4a3aef87c5548564e725d8a69e1ba7c4/org.mixxx.Mixxx.yaml#L23
and if this is not created at install it will probably cause issues when trying to mount it later if you use any org.freedesktop.LinuxAudio.Plugins

Reply:
You likely don't have any flatpak for audio plugins installed, so that would explain why you can't reproduce.

Reply:
I fixed this since it broke the package for users.

Reply:
thanks